update.go

```diff
@@ -265,19 +248,31 @@ other options are ignored.
 					} else {
 						v = -1
 					}
-					*pair.dest = v
+					*pair.dest = &v
 				}
 			}
 
 			r.Pids.Limit = int64(context.Int("pids-limit"))
 		}
 
-		if *r.Memory.Kernel != 0 || *r.Memory.KernelTCP != 0 {
+		// Fix up values
+		if r.Memory.Limit != nil && *r.Memory.Limit == -1 && r.Memory.Swap == nil {
+			// To avoid error "unable to set swap limit without memory limit"
+			r.Memory.Swap = i64Ptr(0)
+		}
+		if r.CPU.Idle != nil && r.CPU.Shares == nil {
+			// To avoid error "failed to write \"4\": write /sys/fs/cgroup/runc-cgroups-integration-test/test-cgroup-7341/cpu.weight: invalid argument"
+			r.CPU.Shares = u64Ptr(0)
+		}
```

Comment:
I'm not sure about this, need to look more closely. Perhaps those errors are actually valid?

I barely remember that runc should reuse the previously set limit (if set).

Will take a closer look later.

Reply:
What should we do?

Reply:
I'm reworking this commit in #4227, hoping to come with simpler code. Still a WIP, will continue tomorrow.

Reply:
This particular part is covered with https://github.com/opencontainers/runc/pull/4226 (which I'm going to carry in one way or another in #4227).